src/runtime/tailwind.config.cjs

```diff
@@ -0,0 +1,41 @@
+const { resolve } = require('path')
+const { defaultExtractor: createDefaultExtractor } = require('tailwindcss/lib/lib/defaultExtractor.js')
+const { customSafelistExtractor } = require('<%= options.runtimeDir %>/utils/colors')
+const { iconsPlugin, getIconCollections } = require('@egoist/tailwindcss-icons')
+
+const defaultExtractor = createDefaultExtractor({ tailwindConfig: { separator: ':' } })
+
+const optionsIcon = <%= JSON.stringify(options.icons) %>
+const iconsPluginParam = Array.isArray(optionsIcon) ? { collections: getIconCollections(optionsIcon) } : typeof optionsIcon === 'object' ? optionsIcon : {}
+
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  darkMode: 'class',
+  plugins: [
+    require('@tailwindcss/forms')({ strategy: 'class' }),
```

Comment:
A lot of this would be solved by https://github.com/tailwindlabs/tailwindcss/pull/12087, so we wouldn't need to create another template for this config, but Tailwind CSS hasn't released this yet - it seems to be part of Oxide which may take a while from them to release. In any case, this is still a positive change imo (as that PR also points out limitations with non-CJS plugins)